### PR TITLE
Update snsdemo to release-dfx-15

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation

# Changes
* Update snsdemo to a release built with dfx v15
* Update dfx to v15
* Add --yes to those remaining `dfx canister install` calls where do don't do so yet; there are more interactive questions.

# Tests
- See CI

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -643,7 +643,7 @@
         "BINSTALL_VERSION": "1.3.0",
         "DIDC_VERSION": "2023-09-27",
         "CARGO_SORT_VERSION": "1.0.9",
-        "SNSDEMO_RELEASE": "release-2023-11-02",
+        "SNSDEMO_RELEASE": "release-dfx-15",
```
Comment (lmuntaner) [inline dfx.json:646]: Will you change this one after merging: https://github.com/dfinity/snsdemo/pull/262 ?

**Review by lmuntaner — COMMENTED** [review]
Does this PR mean that the networks in `dfx.json` do not work anymore?

File: dfx.json (hunk 1 above)
Comment (bitdivine) [reply dfx.json:646]: Yes, or just merge as is and wait for the next automated update, as that takes less time! :-)

Comment (bitdivine) [reply dfx.json:646]: I have triggered a release now, so this can be updated to `release-2023-11-13` whenever it is done.

**Comment by bitdivine** [comment]
> Does this PR mean that the networks in dfx.json do not work anymore?

Local networks are still supported in `dfx 0.15.1`.

**Review by lmuntaner — APPROVED** [review]
Great! Thanks for the explanations!

File: dfx.json (hunk 1 above)
Comment (bitdivine) [reply dfx.json]: Done

**Pull request merged**